public/js/auth.js

```
@@ -167,3 +167,364 @@ async function verifyUserBeforeAction(actionDescription = 'perform this action')
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                'Authorization': `Bearer ${token}`
+            },
+            body: JSON.stringify({ username })
+        });
+
+        const data = await response.json();
+
+        if (response.ok && data.success) {
+            userProfile.username = username;
```

ESLint

[error] 'userProfile' is not defined.

(no-undef)